presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -448,6 +449,88 @@ public static Slice trim(@SqlType("varchar(x)") Slice slice)
         return SliceUtf8.trim(slice);
     }
 
+    @Description("remove the longest string containing only given characters from the beginning of a string")
+    @ScalarFunction("ltrim")
+    @LiteralParameters("x")
+    @SqlType("varchar(x)")
+    public static Slice leftTrim(@SqlType("varchar(x)") Slice slice, @SqlType(StandardTypes.VARCHAR) Slice charactersToTrim)
+    {
+        int[] codePointsToTrim = toCodePoints(charactersToTrim);
+        int start = leftTrimPosition(slice, codePointsToTrim);
+        return slice.slice(start, slice.length() - start);
+    }
+
+    private static int leftTrimPosition(@SqlType("varchar(x)") Slice slice, int[] codePointsToTrim)
+    {
+        int start = 0;
+        SLICE_LOOP:
+        while (start < slice.length()) {
+            int codePoint = getCodePointAt(slice, start);
+            for (int codePointToTrim : codePointsToTrim) {
+                if (codePoint == codePointToTrim) {
+                    start += lengthOfCodePoint(start);
+                    continue SLICE_LOOP;
+                }
+            }
+            break;
+        }
+        return start;
+    }
+
+    private static int[] toCodePoints(Slice slice)
+    {
+        int[] codePointsToTrim = new int[countCodePoints(slice)];
+        int position = 0;
+        for (int index = 0; index < codePointsToTrim.length; index++) {
+            codePointsToTrim[index] = getCodePointAt(slice, position);
+            position += lengthOfCodePoint(slice, position);
+        }
+        return codePointsToTrim;
+    }
+
+    @Description("remove the longest string containing only given characters from the end of a string")
+    @ScalarFunction("rtrim")
+    @LiteralParameters("x")
+    @SqlType("varchar(x)")
+    public static Slice rightTrim(@SqlType("varchar(x)") Slice slice, @SqlType(StandardTypes.VARCHAR) Slice charactersToTrim)
+    {
+        int[] codePointsToTrim = toCodePoints(charactersToTrim);
+        int end = rightTrimPosition(slice, codePointsToTrim);
+        return slice.slice(0, end);
+    }
+
+    private static int rightTrimPosition(@SqlType("varchar(x)") Slice slice, int[] codePointsToTrim)
+    {
+        int end = slice.length();
+        SLICE_LOOP:
+        while (end > 0) {
+            int codePoint = SliceUtf8.getCodePointBefore(slice, end);
```

Comment:
It is probably possible to use SliceUtf8 class in leftTrimPosition as well, maybe we should do that, so the function can look more similar to each other?